chainstate/src/detail/spend_cache.rs

```diff
@@ -201,28 +235,40 @@ impl<'a> CachedInputs<'a> {
                             tx_id: Some(tx.get_id().into()),
                             source_output_index: outpoint.get_output_index() as usize,
                         })?;
-                    output.get_destination().clone()
+                    verify_signature(output.get_destination(), tx, input_idx)
+                        .map_err(|_| BlockError::SignatureVerificationFailed(tx.get_id()))?;
                 }
-                SpendablePosition::BlockReward(_reward_pos) => {
+                SpendablePosition::BlockReward(reward_pos) => {
                     // TODO(Roy): fill this with the block reward that the user now is spending
-                    todo!()
+
+                    let block = self
+                        .db_tx
+                        .get_block(reward_pos.clone())
+                        .map_err(BlockError::from)?
+                        .ok_or(BlockError::NotFound)?;
```

Comment:
Please. Do not use generic error codes. This is extremely dangerous. Errors hould have a context, not just "NotFound" or "Orphan" or something similar. When someone submits a block for processing, returning an error must mean something. What does it mean to P2P when you tell them that the error for the block submitted by a peer is "NotFound"? It means nothing. I got rid of all the Unknown errors which meant nothing, and now we're having another dimension of meaningless errors appearing. This is basically not a NotFound. This is basically an InvalidBlock of some sorts, because the creator of the block referenced a block reward that could not be found in the history. Please give error codes more thought because they're very essential in block processing, or otherwise we'll be open to all kinds of attacks.